include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -247,95 +230,164 @@ struct TypeInfo;
 struct Value;
 struct ModelMetadata;
 
+/** \brief The Env (Environment)
+*
+* The Env holds the logging state used by all other objects.
+* <b>Note:</b> One Env must be created before using any other Onnxruntime functionality
+*/
 struct Env : Base<OrtEnv> {
-  Env(std::nullptr_t) {}
+  explicit Env(std::nullptr_t) {} ///< \brief Create an empty Env object, must be assigned a valid one to be used 
+
+  /** \brief Creates an Env
+  *
+  * Wraps OrtApi::CreateEnv
+  *
+  * \param[in] logging_level
+  * \param[in] logid
+  */
   Env(OrtLoggingLevel logging_level = ORT_LOGGING_LEVEL_WARNING, _In_ const char* logid = "");
+
+  /** \brief Creates and Env
+  *
+  * Wraps OrtApi::CreateEnvWithCustomLogger
```

Comment:
>CreateEnvWithCustomLogger [](http://example.com/codeflow?start=18&length=25)

looks like this one wraps the thread pool CreateEnv instead? similar comment for the next function

Reply:
Ah, they're in a different order to the definitions. I fixed it and reordered them. There are too many ways to create an env.